flow360/plugins/report/uvf_shutter.py

```diff
@@ -572,56 +573,79 @@ class Shutter(Flow360BaseModel):
-                    f"request to shutter: {json.dumps(shutter_request)}"
+
+        async def _build_image_from_local(uvf_request: list[dict], output_folder: str) -> list[str]:
+            input_folder = os.path.join(output_folder, 'input')
+            os.makedirs(input_folder, exist_ok=True)
+
+            log.info(f"dumping the uvf request {uvf_request} to local folder {input_folder}/sequence.json")
+            with open(os.path.join(input_folder, 'sequence.json'), 'w') as json_file:
+                json.dump(uvf_request, json_file, indent=4)
+
+            sequence_file_path = os.path.join(input_folder, 'sequence.json')
+            with open(sequence_file_path, 'r') as file:
+                # Read the content of the file
+                file_content = file.read()
+
+                # Print the content of the file
+                log.debug(f"generate the sequence.json file: {file_content}")
+
+            generated_folder = os.path.join(output_folder, 'shutter_generated')
+            # keep the folder structure as same as s3 storage
+            # mainfest_folder has all of bin files and manifest.json
+            manifest_folder = os.path.join(output_folder,'visualize/manifest')
+            docker_command = [
+                'docker', 'run', '--init',
+                '--gpus', 'all',
+                '-v', manifest_folder + ':/usr/src/app/apps/sandbox/dist/input/:ro',
+                '-v', generated_folder + ':/output/:rw',
+                '-v', sequence_file_path + ':/sequence.json',
+                'shutter', 'sequence', 'run', '/sequence.json'
+            ]
+            log.debug(f"execute the docker command: {' '.join(docker_command)}" )
+            result = subprocess.run(docker_command, shell=False, check=True)
+            log.info(f"completed the docker run with return :{result.returncode}")
+
+
+            # List all entries in the directory
+            entries = os.listdir(generated_folder)
+
+            # Filter out directory names, keeping only files
+            file_names = [entry for entry in entries if os.path.isfile(os.path.join(generated_folder, entry))]
+
+            for file_name in file_names:
+                # Define the full file path
+                source_file = os.path.join(generated_folder, file_name)
+                destination_file = os.path.join(output_folder, file_name)
+
+                # Move the file
+                shutil.move(source_file, destination_file)
+                log.debug(f"Moved {source_file} to {destination_file}")
+
+            return file_names
+
+        tasks = []
+        for case_id, img_folder, uvf_request in screenshots:
+            os.makedirs(img_folder, exist_ok=True)
+            tasks.append(
+                _build_image_from_local(
+                    uvf_request=uvf_request,
+                    output_folder=img_folder
                 )
+            )
+
+
+        responses = await asyncio.gather(*tasks, return_exceptions=True)
```

Comment:
Ohhh is this how your preventing multiple jobs running you just have an await at the end?  If you want it to run async then this will work but you should probably just not have it be an async function then.

Reply:
responses = await asyncio.gather(*tasks, return_exceptions=True)
@maciej-flexcompute is the person to put it as async call. since he is in vacation. I would like to keep the change no change. we could raise the question back when @maciej-flexcompute merge it back to main branch.

Reply:
this could use some work but if you just want to get it out we can leave it but i would put some todos here that this is needs to get cleaned up. This isn't really async call from what i see

Reply:
with rest API calls, it was async meaning every case could be processed in parallel depending how docker instance scales. With CLI, I don't know